includes/express-checkout/class-wc-payments-express-checkout-ajax-handler.php

```diff
@@ -606,4 +606,76 @@ private function transform_ece_address_postcode_data( $address ) {
 
 		return $address;
 	}
+
+	/**
+	 * Modify country locale settings to handle express checkout address requirements.
+	 *
+	 * @param array $locales Array of country locale settings.
+	 * @return array Modified locales array.
+	 */
+	public function modify_country_locale_for_express_checkout( $locales ) {
+		// Only modify locale settings if this is an express checkout AJAX request.
+		if ( ! $this->is_express_checkout_context() ) {
+			return $locales;
+		}
+
+		// For countries that don't have state fields, make the state field optional.
+		foreach ( \WCPay\Constants\Express_Checkout_Element_States::COUNTRIES_WITHOUT_STATES as $country_code ) {
+			$locales[ $country_code ]['state']['required'] = false;
+		}
+
+		return $locales;
+	}
+
+	/**
+	 * Check if we're in an express checkout context.
+	 *
+	 * @return bool True if we're in an express checkout context, false otherwise.
+	 */
+	private function is_express_checkout_context() {
+		// Only proceed if this is a Store API request.
+		if ( ! WC_Payments_Utils::is_store_api_request() ) {
+			return false;
+		}
+
+		// Check the payment data in the request.
+		$request_body = file_get_contents( 'php://input' );
+		$request_data = json_decode( $request_body, true );
+
+		if ( ! isset( $request_data['payment_data'] ) ) {
+			return false;
+		}
+
+		return $this->is_express_payment_type( $request_data['payment_data'] );
```

Comment:
(minor and optional :) ) one alternative to these checks could be to implement similar checks as the ones defined on top of the `tokenized_cart_store_api_address_normalization` method: https://github.com/Automattic/woocommerce-payments/blob/c8bad250e705bfa62c68037de94bd3332ca25ce9/includes/express-checkout/class-wc-payments-express-checkout-ajax-handler.php#L500-L508

This way, you don't need to inspect the request's payment data, and you can use the nonce validation that is already used in other methods.

Reply:
I think that's a better way thanks for the suggestion. I tried the same approach in this commit [96d5616](https://github.com/Automattic/woocommerce-payments/pull/10770/commits/96d561697567a38fa649d9e111c6c629f36a05a0). The only difference is that   we don't have direct access to the request in the `woocommerce_get_country_locale` filter, so I'm using superglobals to access it. Please have a look and let me know what you think.

Reply:
Using the superglobals seems fine, since `$request` isn't filled yet. We need to use those same superglobals in `WC_Payments_Payment_Request_Session` for the same reason :)